packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -685,4 +828,43 @@ identityField: "${rule.identityField || DEFAULT_IDENTITY_FIELD}" }`
     public setUserGroups(): SetNode {
         return set(ref('userGroups'), raw('$util.defaultIfNull($ctx.identity.claims.get("cognito:groups"), [])'));
     }
-}
+
+    public setCustomClaim(customGroups: string): SetNode {
```

Comment:
This method is problematic as you are setting userGroups to an unknown type, while setUserGroups uses cognito:groups which is known to be an array type. userGroups should have a stable type. Ive left a comment above on how this method can be removed completely.

Reply:
Added this check in setUsergroups where it'll reassign the the string into an array